### PR TITLE
Start using golangci-lint instead of the deprecated x/lint.

### PR DESCRIPTION
**Pull request by aalexand**
See https://github.com/golang/go/issues/38968 for the x/lint deprecation message.

I'm also putting all tool invocations into test.sh so that it's easier to run all checks locally during the development time. GitHub Actions are nice but they make running the checks locally more difficult.

### CODE REVIEW

File: profile/legacy_profile.go [hunk 1]
```diff
@@ -865,7 +865,6 @@ func parseThread(b []byte) (*Profile, error) {
 	// Recognize each thread and populate profile samples.
 	for !isMemoryMapSentinel(line) {
 		if strings.HasPrefix(line, "---- no stack trace for") {
-			line = ""
```
Comment (wyk9787) [inline profile/legacy_profile.go:868]: Is this intentional?

File: test.sh [hunk 2]
```diff
@@ -40,3 +40,7 @@ for d in $PKG; do
   fi
 done
 
+go vet -all ./...
+golangci-lint run -D errcheck ./...  # TODO: Enable errcheck back.
```
Comment (wyk9787) [inline test.sh:44]: I assume we have places in the pprof CLI that we don't check errors so we need to disable this in the linter? Are you  going to fix it separately?

**Review by wyk9787 — CHANGES_REQUESTED** [review]

File: test.sh (hunk 2 above)
Comment (aalexand) [reply test.sh:44]: Yes, the linter complained about a few places there. I don't plan to work on this soon as it's not a regression.

File: profile/legacy_profile.go (hunk 1 above)
Comment (aalexand) [reply profile/legacy_profile.go:868]: Yes, it fixed a lint warning that this assignment is redundant - we break out of the loop and this var is not used after the loop.

Comment (wyk9787) [reply profile/legacy_profile.go:868]: Ah, okay. Thanks.

**Review by wyk9787 — APPROVED** [review]

**Review by ghemawat — COMMENTED** [review]
LGTM

**Comment by Labutin** [comment]
golangci-lint run -D errcheck ./... fails for my pull request https://github.com/google/pprof/pull/734
See an example: https://github.com/google/pprof/actions/runs/3422932933/jobs/5700946312
Is it expected?

**Comment by aalexand** [comment]
Not expected. Looks like it fails at Go tip - perhaps Go mainline added something that the linter doesn't like.

**Pull request merged**